src/rules/sortImportsRule.ts

```diff
@@ -0,0 +1,246 @@
+        this.addFailureAtNode(
+          node,
+          `All imports of the same type must be sorted alphabetically. "${importData.sortValue}" must come before "${this.currentSortValue.originalValue}"`);
+      } else {
+        this.currentSortValue = {
+          sortValue: this.caseConverter(importData.sortValue),
+          originalValue: importData.sortValue};
+      }
+    } else {
+      const currentSyntaxType = MemberSyntaxType[importData.memberSyntaxType];
+      const previousSyntaxType = MemberSyntaxType[this.expectedOrder[this.currentImportIndex]];
+      this.addFailureAtNode(
+        node,
+        `All imports of type "${currentSyntaxType}" must occur before all imports of type "${previousSyntaxType}"`);
+    }
+  }
+
+  private _determineImportType(node: ts.ImportDeclaration | ts.ImportEqualsDeclaration): ImportMetadata {
+    const nodeText = node.getFullText();
+
+    if (node.kind === ts.SyntaxKind.ImportEqualsDeclaration) {
+      const aliasMatch = /\bimport\s+(\w+)\s*=.+/g.exec(nodeText);
+      return {
+        memberSyntaxType: MemberSyntaxType.Alias,
+        sortValue: aliasMatch[1]
+      };
+    } else {
+      const singleMatch = /\bimport\s+({?([^,{}\*]+?)}?)\s*from\s+[\'"](?:[^"\']+)["\']/g.exec(nodeText);
+      const multipleMatch = /\bimport\s*{?\s*([^{}\'",]+?)\s*,(?:\s*.+\s*,\s*)*\s*.+\s*}?\s*from\s+[\'"](?:[^"\']+)["\']/g.exec(nodeText);
+      const noneMatch = /\bimport\s+[\'"]([^"\']+)["\']/g.exec(nodeText);
+      const allMatch = /\bimport\s+\*\s+as\s+(.+)\s+from\s+[\'"](?:[^"\']+)["\']/g.exec(nodeText);
+
+      if (singleMatch !== null) {
+        return {
+          memberSyntaxType: MemberSyntaxType.Single,
+          sortValue: singleMatch[1]
+        };
+      } else if (multipleMatch !== null) {
+        return {
+          memberSyntaxType: MemberSyntaxType.Multiple,
+          sortValue: multipleMatch[1]
+        };
+      } else if (noneMatch !== null) {
+        return {
+          memberSyntaxType: MemberSyntaxType.None,
+          sortValue: noneMatch[1]
+        };
+      } else if (allMatch !== null) {
+        return {
+          memberSyntaxType: MemberSyntaxType.All,
+          sortValue: allMatch[1]
+        };
+      }
+      else {
+        this.addFailureAtNode(node, 'Could not determine import type');
+      }
+    }
+  }
+
+  private _processMemberSyntaxSortOrder(sortOption: string[]): MemberSyntaxType[] {
+    const defaultOrder = [MemberSyntaxType.None, MemberSyntaxType.All, MemberSyntaxType.Multiple, MemberSyntaxType.Single, MemberSyntaxType.Alias];
+    if (Array.isArray(sortOption) && typeof sortOption[0] === 'string' && sortOption.length === 5) {
+      const order: MemberSyntaxType[] = [];
+      const usedOptions = {};
+      sortOption.forEach((t) => {
+        if (usedOptions[t] !== undefined) {
+          // Warning: we have seen this one already - skip
+        } else {
+          usedOptions[t] = t;
+          switch (t) {
```

Comment:
Instead of this switch a simple map could be created:

```ts
const memberSyntaxTypeMap = {
  none: MemberSyntaxType.None,
  all: MemberSyntaxType.All,
  // rest go here
};
if (memberSyntaxTypeMap[t]) {
  order.push(memberSyntaxTypeMap[t]);
}
```